common/txmgr/types/client.go

```diff
@@ -30,6 +31,7 @@ type TxmClient[
 		ctx context.Context,
 		attempts []TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE],
 	) (txReceipt []R, txErr []error, err error)
+	HeadByHash(ctx context.Context, hash BLOCK_HASH) (HEAD, error)
```

Comment:
I think evm TXM client uses the general Client evm interface, so this is not needed.